### PR TITLE
Contact Requests overhaul

### PR DESCRIPTION
**Pull request by cyBerta**
depending on https://github.com/deltachat/deltachat-core-rust/pull/2514 to be merged

other than that ready for review

### CODE REVIEW

**Comment by r10s** [comment]
did not go through the code yet, but a little layout comment:

i assume contact requests are not notified? if so, the badge of contact requests should not be _red_ but in the _grey_ of muted chats.

File: deltachat-ios/Chat/ChatViewController.swift [hunk 1]
```diff
@@ -266,15 +274,20 @@ class ChatViewController: UITableViewController {
         configureEmptyStateView()
 
         if !disableWriting {
-            configureMessageInputBar()
-            draft.parse(draftMsg: dcContext.getDraft(chatId: chatId))
-            messageInputBar.inputTextView.text = draft.text
-            configureDraftArea(draft: draft, animated: false)
-            editingBar.delegate = self
```
Comment (r10s) [inline deltachat-ios/Chat/ChatViewController.swift:273]: `editingBar.delegate = self`  is not needed in `configureUIForWriting()` as it was always already done in `lazy var editingBar`, right?

File: deltachat-ios/Controller/MailboxViewController.swift [hunk 2]
```diff
@@ -1,68 +0,0 @@
-import UIKit
```
Comment (r10s) [inline deltachat-ios/Controller/MailboxViewController.swift:1]: cool that we never renamed that to `ContactRequestsViewController` - would have been a waste of time :)

”Wisdom of a Stone" :)

**Review by r10s — APPROVED** [review]
lgtm, there are some minor things to be done, but that should better go to subsequent prs to have a ios-master working again with core-master.

File: deltachat-ios/Chat/ChatViewController.swift (hunk 1 above)
Comment (cyBerta) [reply deltachat-ios/Chat/ChatViewController.swift:273]: that's right

File: deltachat-ios/Controller/MailboxViewController.swift (hunk 2 above)
Comment (cyBerta) [reply deltachat-ios/Controller/MailboxViewController.swift:1]: 🤣

**Pull request merged**